polling.lua

```diff
@@ -36,17 +36,25 @@ bot.init()
 
 api.firstUpdate()
 while true do -- Start a loop while the bot should be running.
-	local res = api.getUpdates(last_update+1) -- Get the latest updates
-	if res then
-		-- clocktime_last_update = os.clock()
-		for i=1, #res.result do -- Go through every new message.
-			last_update = res.result[i].update_id
-			--print(last_update)
-			current.h = current.h + 1
-			main.parseMessageFunction(res.result[i])
+	local status, err = pcall(
+	function ()
+		local res = api.getUpdates(last_update+1) -- Get the latest updates
+		if res then
+			-- clocktime_last_update = os.clock()
+			for i=1, #res.result do -- Go through every new message.
+				last_update = res.result[i].update_id
+				--print(last_update)
+				current.h = current.h + 1
+				processUpdate(res.result[i])
+			end
+		else
+			print('Connection error')
 		end
-	else
-		print('Connection error')
+	end)
+	if err then 
+		print(err)
+		api.sendLog(err)
```

Comment:
If this error happens because of issues with API, isn’t it better to not try to use the api to send this log? Otherwise seems fine to me.

Reply:
I don't think it hurts? worst that happens is infinite recursion and a crash. Although that could be detected too of course.

Reply:
Yeah, it can get very spammy with the error log function giving another error while trying to log the previous error...